packages/moonstone/BodyText/BodyText.js

```diff
@@ -48,18 +64,49 @@ const BodyTextBase = kind({
 		 * @type {Object}
 		 * @public
 		 */
-		css: PropTypes.object
+		css: PropTypes.object,
+
+		/**
+		 * Toggles multi-line (`false`) vs single-line (`true`) behavior. `noWrap` mode
+		 * automatically enables {@link moonstone/Marquee} so long text isn't permanently occluded.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @public
+		 */
+		noWrap: PropTypes.bool
+	},
+
+	defaultProps: {
+		noWrap: false
 	},
 
 	styles: {
 		css: componentCss,
 		publicClassNames: 'bodyText'
 	},
 
-	render: ({css, ...rest}) => {
+	computed: {
+		className: ({noWrap, styler}) => styler.append({noWrap})
+	},
+
+	render: ({centered, css, noWrap, ...rest}) => {
+		if (noWrap) {
+			return (
+				<MarqueeBodyText
+					component="div" // Assign a new component to BodyText, since DIV is not allowed inside a P tag (the default for BodyText)
+					marqueeOn="render"
```

Comment:
Not suggesting to change, but I'm concerned if there's a use case for a `hover`.

Reply:
`rest` is spread after this is applied, so this will be overridden if the developer specifies `marqueeOn="hover" or any other marqueeOn values.

Reply:
Heh, that's a bit sneaky. The concern is whether developers want to do that and if so should it be documented? Or description in `noWrap` good enough?

We can follow up later if the use case comes up later. I think supporting marquee on hover for non-spottable component is weird IMO.

Reply:
I think the existing description is satisfactory. It references and links to marquee, so (to me at least) the marquee functionality, and by extension its props, would be available. Though, you're right, it's not explicit.